manager/engineimage.go

```diff
@@ -100,8 +100,8 @@ func (m *VolumeManager) DeleteEngineImageByName(name string) error {
 	return nil
 }
 
-func (m *VolumeManager) DeployAndWaitForEngineImage(image string) error {
-	if _, err := m.GetEngineImage(image); err != nil {
+func (m *VolumeManager) DeployEngineImage(image string) error {
+	if _, err := m.GetEngineImageByImage(image); err != nil {
```

Comment:
I find this pattern easier to read, up to you, what do you think?
This reduces a level of nesting and moves the real error & exit to the top before proceeding with expected results (NotFound)

```
if !datastore.ErrorIsNotFound(err) {
    return return errors.Wrapf(err, "cannot create engine image for %v", image)
}

if _, err = m.CreateEngineImage(image); err != nil {
	return errors.Wrapf(err, "cannot create engine image for %v", image)
}
```

Reply:
Your recommendation is better. Fixed. Thank you